pkg/api/hephaestus/v1/imagebuild_webhook.go

```diff
@@ -42,12 +42,16 @@ func (in *ImageBuild) validateImageBuild(action string) (admission.Warnings, err
 	var errList field.ErrorList
 	fp := field.NewPath("spec")
 
-	if strings.TrimSpace(in.Spec.Context) == "" {
-		log.V(1).Info("Context is blank")
-		errList = append(errList, field.Required(fp.Child("context"), "must not be blank"))
-	} else if _, err := url.ParseRequestURI(in.Spec.Context); err != nil {
-		log.V(1).Info("Context is not a valid URL")
-		errList = append(errList, field.Invalid(fp.Child("context"), in.Spec.Context, err.Error()))
+	if strings.TrimSpace(in.Spec.Context) == "" && strings.TrimSpace(in.Spec.DockerfileContents) == "" {
+		log.V(1).Info("Context and DockerfileContents are both blank")
+		errList = append(errList, field.Required(fp.Child("context or dockerFileContents"), "must not be blank"))
```

Comment:
this should probably be two separate errors. with `fp.Child`, it looks like:
```
The ImageBuild "dkrrm56j" is invalid: spec.context or dockerFileContents: Required value: must not be blank
```

Reply:
Ok I changed this around to make it more correct and say:
`The ImageBuild "dkrrm56j" is invalid: spec.context: Required value: must not be blank if spec.dockerfileContents is blank`